x-pack/plugins/apm/public/components/shared/charts/transaction_breakdown_chart/transaction_breakdown_chart_contents.tsx

```diff
@@ -51,24 +51,15 @@ export function TransactionBreakdownChartContents({
   timeseries,
 }: Props) {
   const history = useHistory();
-  const chartRef = React.createRef<Chart>();
   const { annotations } = useAnnotationsContext();
   const chartTheme = useChartTheme();
-  const { pointerEvent, setPointerEvent } = useChartPointerEventContext();
+  const { chartRef, setPointerEvent } = useChartPointerEventContext(
+    'timeSpentBySpan'
+  );
   const { urlParams } = useUrlParams();
   const theme = useTheme();
   const { start, end } = urlParams;
 
-  useEffect(() => {
-    if (
-      pointerEvent &&
-      pointerEvent.chartId !== 'timeSpentBySpan' &&
-      chartRef.current
-    ) {
-      chartRef.current.dispatchExternalPointerEvent(pointerEvent);
-    }
-  }, [chartRef, pointerEvent]);
```

Comment:
@cauemarcondes Do you agree that it makes sense to co-locate this logic with the `useChartPointerEventContext`?
Afaict it's always used in combination with said hook, so co-locating seems like the right approach.

Reply:
Why do we need this distinction for `timeSpentBySpanType`?

Reply:
Yes, thanks for doing it

Reply:
> Why do we need this distinction for timeSpentBySpanType?

What distinction are you referring to?

Reply:
I was mostly wondering why we need to filter out a chart's own events. Does it cause some kind of loop where the `onPointerUpdate` cb is continuously called? Asking because there's seems to be some kind of `isEquals` check before firing the `onPointerUpdate` callback: https://github.com/elastic/elastic-charts/blob/d28820858d013326b3c660381e70696e9382166d/src/chart_types/xy_chart/state/selectors/on_pointer_move_caller.ts#L128-L130

Reply:
Also other teams don't seem to filter out a chart's own events either:
`x-pack/plugins/infra/public/pages/metrics/inventory_view/components/node_details/tabs/metrics/metrics.tsx`
`src/plugins/vis_type_timeseries/public/application/visualizations/views/timeseries/index.js`

Reply:
@dgieselaar good catch, it looks like it's not needed. https://elastic.github.io/elastic-charts/?path=/story/interactions--cursor-update-action

Reply:
Nice, I'll update.